build/templates/rep_caps.rst.mako

```diff
@@ -19,16 +19,20 @@
 
 ${helper.get_rst_header_snippet('Repeated Capabilities', '=')}
 
-    Repeated capabilities attributes are used to set the `channel_string` parameter to the
-    underlying driver function call. This can be the actual function based on the :py:class:`Session`
-    method being called, or it can be the appropriate Get/Set Attribute function, such as :c:`${config['c_function_prefix']}SetAttributeViInt32()`.
+    :py:class:`${module_name}.Session` supports "Repeated Capabilities", which are multiple instances of the same type of
+    functionality. The repeated capabilities supported by :py:class:`${module_name}.Session` are:
 
-    Repeated capabilities attributes use the indexing operator :python:`[]` to indicate the repeated capabilities.
-    The parameter can be a string, list, tuple, or slice (range). Each element of those can be a string or
-    an integer. If it is a string, you can indicate a range using the same format as the driver: :python:`'0-2'` or
-    :python:`'0:2'`
+% for rep_cap in config['repeated_capabilities']:
+<%
+name = rep_cap['python_name']
+%>\
+    #. ${name}_
+% endfor
+
+    Use the indexing operator :python:`[]` to indicate which repeated capability you are trying to access.
+    The parameter can be an integer, a string, a list, a tuple, or slice (range).
```

Comment:
Few comments:
1) `slice` and `range` are two different things. `slice` is a built-in [function](https://docs.python.org/3/library/functions.html#slice) that returns a slice object. `range` is a [type](https://docs.python.org/3/library/functions.html#func-range).
2) `list`, `tuple`, and `range` are collectively known as [basic sequence types](https://docs.python.org/3/library/stdtypes.html#typesseq).
3) We support a range-based string using a dash or a colon. e.g. `session.channels['1-4']` and `session.channels['1:4']`. These are just strings as far as the language is concerned. But we have [logic](https://github.com/ni/nimi-python/blob/f074be747bf32054fce5ffe13047a9412806030b/generated/nifake/nifake/_converters.py#L66) to handle these.

We can be explicit about the types we support, along with an example for each:
- `int`
  - `session.foo[42]`
- `str`
  - Single instance: `session.foo['42']`
  - A range of instances: `session.foo['0-2']` or `session.foo['0:2']`
- `list`
  - `session.foo[0,1,2]`
- `tuple`
  - `session.foo[(0,1,2)]`
- `range`
  - `session.foo[range(10)]`
- `slice`
  - `session.foo[range(10)[3:5]]`

Reply:
Note that the APIs will work with any iterable.

Reply:
>Note that the APIs will work with any iterable.
Yes, the APIs will, though it seems like there are certain repeated capabilities that will only accept strings?
instruments?
maybe pins?

>We can be explicit about the types we support, along with an example for each:
@sbethur are you talking about the example code shown in the individual sections for each repeated capability or new examples that should be placed before the individual sections for each repeated capability type? If it's the former, that will further complicate an already difficult task.

Reply:
Well, not all iterables - `dict` and `set` won't work.

Reply:
> new examples that should be placed before the individual sections 

This

Reply:
Elsewhere Marcos suggested not documenting range-based strings, which I think is a good idea. If we leave that out, then the supported types are all Python-native and we don't need to elaborate the support with examples. Here's my new suggestion:

> The parameter can be a single element or an iterable that implements sequence semantics, like list, tuple, range, and slice.